CONTRIBUTING.md

```diff
@@ -1,79 +1,232 @@
+## Contributing to AsyncAPI Libraries
+
+A common point of confusion for those who wish to contribute to AsyncAPI is where
+to start. In fact, you may have found yourself here after attempting to make an
+improvement to an AsyncAPI library. Should a new addition be made to the AsyncAPI
+spec first or an AsyncAPI library first? Admittedly, this can become a bit of a
+[chicken-or-egg](https://en.wikipedia.org/wiki/Chicken_or_the_egg) dilemma.
+
+AsyncAPI libraries seek to be "spec compliant", which means they discourage
+changes that cause them to behave differently from the spec as written. However,
+they also encourage pull requests for changes that accompany an RFC *proposal*
+or RFC *draft*. In fact, a spec contribution RFC won't be *accepted* until it
+has experience being implemented in an AsyncAPI library.
+
+To allow a library to remain spec compliant while also implementing *proposals*
+and *drafts*, the library's maintainers may request that these new features are
+disabled by default with opt-in option flags or they may simply wait to merge a
+well-tested pull request until the spec proposal is *accepted*.
+
+
+## Guiding Principles
+
+AsyncAPI's evolution is guided by a few principles. Suggested contributions
+should use these principles to guide the details of an RFC and decisions to
+move forward.
+
+* **Favor no change**
+
+  As AsyncAPI is implemented in many languages under the collaboration
+  of a lot of individuals, incorporating any change has a high cost.
+  Accordingly, proposed changes must meet a very high bar of added value.
+  The burden of proof is on the contributor to illustrate this value.
+
+* **Enable new capabilities motivated by real use cases**
+
+  Every change should intend on unlocking a real and reasonable use case. Real
+  examples are always more compelling than theoretical ones, and common
+  scenarios are more compelling than rare ones. RFCs should do more than offer
+  a different way to reach an already achievable outcome.
+
+* **Simplicity and consistency over expressiveness and terseness**
+
+  "Possible but awkward" is often favored over more complex
+  alternatives. Simplicity (e.g. fewer concepts) is more important than
+  expressing more sophisticated ideas or writing less.
+
+* **Preserve option value**
+
+  It's hard to know what the future brings; whenever possible, decisions should
+  be made that allow for more options in the future. Sometimes this is
+  unintuitive: spec rules often begin more strict than necessary with a future
+  option to loosen when motivated by a real use case.
+
+* **Understandability is just as important as correctness**
+
+  The AsyncAPI spec, despite describing technical behavior, is intended to be
+  read by people. Use natural tone and include motivation and examples.
+
+
```

Comment:
everything below this comment can be removed. When I see it in this PR I kinda remind myself with spec and it makes me feel we do not make anything easy and just as complicated as in spec.

so just add another paragraph in Guiding section:

```
Pull request with new or improved extension will be merged only if there is compliant implementation in the the [AsyncAPI JSON Schema](https://github.com/asyncapi/spec-json-schemas), [AsyncAPI JS Parser](https://www.github.com/asyncapi/parser-js) and the [AsyncAPI React component](https://github.com/asyncapi/asyncapi-react).

You do not have to fulfil all the requirements upon pull request creation. First focus on documentation and a use case. Once maintainers of the catalog tell you that it is the right time to work on tooling implementation, then start doing it or fund volunteers that can help.
```

later once we have that figured in ☝🏼 mentioned repos, we can extend guid with links to proper instructions on how to do it.

Reply:
okay